samples/cpp/hello_nv12_input_classification/main.cpp

```diff
@@ -53,258 +61,102 @@ std::pair<size_t, size_t> parseImageSize(const std::string& size_string) {
-            assert(netInputSize == inputs.size());
-            std::cout << "Infer using BatchedBlob of NV12 images." << std::endl;
-            Blob::Ptr batched_input = make_shared_blob<BatchedBlob>(inputs);
-            inputs = {batched_input};
-        }
-
-        /** Read labels from file (e.x. AlexNet.labels) **/
-        std::string labelFileName = fileNameNoExt(input_model) + ".labels";
+        // -------- Step 1. Initialize OpenVINO Runtime Core ---------
+        ov::runtime::Core core;
+
+        // -------- Step 2. Read a model --------
+        slog::info << "Loading model files: " << model_path << slog::endl;
+        auto model = core.read_model(model_path);
+        printInputAndOutputsInfo(*model);
+
+        OPENVINO_ASSERT(model->get_parameters().size() == 1, "Sample supports models with 1 input only");
+        OPENVINO_ASSERT(model->get_results().size() == 1, "Sample supports models with 1 output only");
+
+        std::string input_tensor_name = model->input().get_any_name();
+        std::string output_tensor_name = model->output().get_any_name();
+
+        // -------- Step 3. Add preprocessing  --------
+        // clang-format off
+        model = PrePostProcessor(model).
+            // 1) Select input with 'input_tensor_name' tensor name
+            input(InputInfo(input_tensor_name).
+                // 2) Set input type
+                // - as 'u8' precision
+                // - set color format to NV12 (single plane)
+                // - static spatial dimensions for resize preprocessing operation
+                tensor(InputTensorInfo().
+                    set_element_type(ov::element::u8).
+                    set_color_format(ColorFormat::NV12_SINGLE_PLANE).
+                    set_spatial_static_shape(input_height, input_width)).
+                // 3) Pre-processing steps:
+                //    a) Convert to 'float'. This is to have color conversion more accurate
+                //    b) Convert to RGB: Assumes that model accepts images in RGB format.
+                //       For BGR, change it manually
+                //    c) Convert layout to model's one. It is done before 'resize' in this
+                //       sample, as there can be plugin limitation with support of resize in
+                //       NHWC format
+                //    d) Resize image from tensor's dimensions to model ones
+                preprocess(PreProcessSteps().
+                    convert_element_type(ov::element::f32).
+                    convert_color(ColorFormat::RGB).
+                    convert_layout({"NCHW"}).
+                    resize(ResizeAlgorithm::RESIZE_LINEAR)).
+                // 4) Set model data layout (Assuming model accepts images in NCHW layout)
+                network(InputNetworkInfo().
+                    set_layout("NCHW"))).
+            // 5) Apply preprocessing to a input with 'input_tensor_name' name of loaded model
+            build();
+        // clang-format on
```

Comment:
If 'chaining' style looks too complicated, consider use 'getters' API. Other important changes in this suggestion:
- Convert to BGR instead of RGB
- Removed explicit 'convert_layout' from preprocess steps as CPU supports resizing 'NHWC' now
 
```suggestion
        // -------- Step 3. Add preprocessing  --------
        PrePostProcessor preproc = PrePostProcessor(model);
        // 1) Select input with 'input_tensor_name' tensor name
        InputInfo& input_info = preproc.input(input_tensor_name);
        // 2) Set input type
        // - as 'u8' precision
        // - set color format to NV12 (single plane)
        // - static spatial dimensions for resize preprocessing operation
        input_info.tensor()
            .set_element_type(ov::element::u8)
            .set_color_format(ColorFormat::NV12_SINGLE_PLANE)
            .set_spatial_static_shape(input_height, input_width);
        // 3) Pre-processing steps:
        //    a) Convert to 'float'. This is to have color conversion more accurate
        //    b) Convert to BGR: Assumes that model accepts images in BGR format. For RGB, change it manually
        //    c) Resize image from tensor's dimensions to model ones
        input_info.preprocess()
            .convert_element_type(ov::element::f32)
            .convert_color(ColorFormat::BGR)
            .resize(ResizeAlgorithm::RESIZE_LINEAR);
        // 4) Set model data layout (Assuming model accepts images in NCHW layout)
        input_info.network().set_layout("NCHW");
        // 5) Apply preprocessing to an input with 'input_tensor_name' name of loaded model
        model = preproc.build();
```